pkg/operator/helpers.go

```diff
@@ -4,9 +4,39 @@ package operator
 // Licensed under the Apache License 2.0.
 
 import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
 	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
 func GatewayEnabled(cluster *arov1alpha1.Cluster) bool {
 	return len(cluster.Spec.GatewayDomains) > 0
 }
+
+// PodSecurityAdmissionControl is an admissions controller
+// for pods which replaces pod security policies, enabled on
+// OpenShift 4.11 and up
+func PodSecurityAdmissionControl(ctx context.Context, client client.Client) (bool, error) {
+	cv := &configv1.ClusterVersion{}
+
+	err := client.Get(ctx, types.NamespacedName{Name: "version"}, cv)
+	if err != nil {
+		return false, err
+	}
+
+	vers, err := version.ParseVersion(cv.Spec.DesiredUpdate.Version)
```

Comment:
This should be as below. we want to get the current cluster version instead of the desired one to prevent a panic.

`vers, err := version.GetClusterVersion(cv)`

Reply:
Not sure if Ben used the _cv.Spec.DesiredUpdate.Version_ on purpose (maybe we need to use it to not apply the privilegedNamespaceLabels once it is too late. If we check for the cluster version once cluster was updated to 4.11+ maybe we have already failed to schedule the pods -but maybe it is not a problem-).

If it was done on purpose I think we need at least a 
```go
if cv.Spec.DesiredUpdate != nil 
```
to avoid a possible panic. 

Another option would be to check first for a possible desired update (with the current code + protection to nil) to 4.11 and if not, check the version with version.GetClusterVersion(cv).

Reply:
Using the correct function now.  Should be good as I dont' see where a panic can happen unless the CV object does not exist entirely, but then `err != nil`

Reply:
Nice. The panic I was referring to was on the direct access to _cv.Spec.DesiredUpdate_, now all safe 🙂